src/main/java/com/crud/tasks/controller/TaskController.java

```diff
@@ -28,22 +28,22 @@
 
     @RequestMapping(method = RequestMethod.GET, value = "tasks/")
     @ResponseBody
-    public TaskDto getTask(@RequestParam Long taskId) throws TaskNotFoundException
-    {
+    public TaskDto getTask(@RequestParam Long taskId) throws TaskNotFoundException {
         return taskMapper.mapToTaskDto(service.getTask(taskId).orElseThrow(TaskNotFoundException::new));
     }
 
-    @RequestMapping(method = RequestMethod.DELETE, value = "tasks/{id}")
+    @RequestMapping(method = RequestMethod.DELETE, value = "tasks")
     @ResponseStatus(HttpStatus.NO_CONTENT)
-    public void deleteTask(@PathVariable("id") Long taskId) {
+    public void deleteTask(@RequestParam Long taskId) {
```

Comment:
why do you use RequestParam instead of PathVariable?

Reply:
Could you also check what happens when you provide not existing id AND invalid id (like 'abc' instead of number) - I am just curious ;)

Reply:
I changed to @RequestParam because i wanted to have coherent class TaskController (according to module 15).
If i provide not existing id OR invidal id, It'll be displayed "status": 500,  "error": "Internal Server Error",  "message": "No class com.crud.tasks.domain.Task entity with id 2 exists!".
Should i throw exception here?

Reply:
error codes 5XX means that there is a problem in server side - so it means your fault, but in this case you should return code 4XX - which means client error. Try to google about exception handling in spring MVC.